daemon/regmaild/regmaild.c

```diff
@@ -100,52 +147,47 @@ int main(int argc, char *argv[])
     if (regmaildb_open(&Db, fpath) != SQLITE_OK)
     {
         fprintf(stderr, "cannot initialize emaildb: %s.\n", fpath);
-        return 0;
+        return 1;
     }
 
-    if (sqlite3_prepare(Db, "REPLACE INTO emaildb (userid, email) VALUES (lower(?),lower(?));",
-                -1, &Stmt, NULL) != SQLITE_OK ||
-        sqlite3_prepare(Db, "BEGIN TRANSACTION;", -1, &tranStart, NULL) != SQLITE_OK ||
+    if (!ops->prepare_stmt(Db, &Stmt))
+        return 1;
+
+    if (sqlite3_prepare(Db, "BEGIN TRANSACTION;", -1, &tranStart, NULL) != SQLITE_OK ||
         sqlite3_prepare(Db, "COMMIT;", -1, &tranEnd, NULL) != SQLITE_OK)
     {
-        fprintf(stderr, "SQLite 3 internal error.\n");
-        return 0;
+        fprintf(stderr, "Error preparing txn stmts: %s\n", sqlite3_errmsg(Db));
+        return 1;
     }
 
+    int error = 0;
     sqlite3_step(tranStart);
     sqlite3_reset(tranStart);
     while (read(fd, &xuser, sizeof(xuser)) == sizeof(xuser))
     {
         i++;
         // got a record
-        if (strlen(xuser.userid) < 2 || strlen(xuser.userid) > IDLEN)
-            continue;
-        if (strlen(xuser.email) < 5)
+        if (!ops->filter_user(&xuser))
             continue;
 
-        if (sqlite3_bind_text(Stmt, 1, xuser.userid, strlen(xuser.userid), 
-                    SQLITE_STATIC) != SQLITE_OK)
+        if (!ops->bind_args(Db, Stmt, &xuser))
         {
-            fprintf(stderr, "\ncannot prepare userid param.\n");
-            break;
-        }
-        if (sqlite3_bind_text(Stmt, 2, xuser.email, strlen(xuser.email), 
-                    SQLITE_STATIC) != SQLITE_OK)
-        {
-            fprintf(stderr, "\ncannot prepare email param.\n");
+            error = 1;
             break;
         }
 
         if (sqlite3_step(Stmt) != SQLITE_DONE)
         {
-            fprintf(stderr, "\ncannot execute statement.\n");
+            error = 1;
+            fprintf(stderr, "\ncannot execute statement: %s\n",
+                    sqlite3_errmsg(Db));
             break;
         }
         sqlite3_reset(Stmt);
 
-        valids ++;
+        valids++;
         if (valids % 10 == 0)
-            fprintf(stderr, "%d/%d (valid: %d)\r", 
```

Comment:
Could you split out the error number fixes and style fixes into separate patches when possible?
That would make it easier to read.